### PR TITLE
Add support for Packer packages

### PR DESCRIPTION
**Pull request by tpdownes**
This PR improves support for remote Packer templates by enabling a package notation with source lines such as

```yaml
- group: slurm
  modules:
  - id: slurm-image
    source: github.com/SchedMD/slurm-gcp//packer?ref=5.7.4&depth=1
    kind: packer
```

This will cause the entire `slurm-gcp` repo to be staged to `slurm/slurm-image` and the source of the Packer module will be treated as `slurm/slurm-image/packer`. The problem being solved is that the Packer module may itself contain local references within the rest of the repo (i.e. `"../scripts"`).

Documentation will be added in a future PR alongside a production-ready example. A "beta" example blueprint is shown below that invokes the DAOS and Slurm image building processes is shown below. It has the following limitations:

- the DAOS module does not allow the selection of network (it defaults to the "default" network which must have IAP tunneling setup properly)

```yaml
---
blueprint_name: remote-packer

vars:
  project_id:  ## Set GCP Project ID Here ##
  deployment_name: images
  region: us-central1
  zone: us-central1-c

# Documentation for each of the modules used below can be found at
# https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/main/modules/README.md

deployment_groups:
- group: network
  modules:
  - id: network0
    source: modules/network/vpc

- group: daos
  modules:
  - id: daos-image
    source: github.com/daos-stack/google-cloud-daos//images?ref=v0.4.0&depth=1
    kind: packer
    settings:
      daos_version: 2.2.0
      daos_repo_base_url: https://packages.daos.io
      daos_packages_repo_file: EL8/packages/x86_64/daos_packages.repo
      use_iap: true
      enable_oslogin: false
      machine_type: n2-standard-32
      source_image_family: hpc-rocky-linux-8
      source_image_project_id: cloud-hpc-image-public
      image_guest_os_features: ["GVNIC"]
      disk_size: "20"
      state_timeout: "10m"
      scopes: ["https://www.googleapis.com/auth/cloud-platform"]
      use_internal_ip: true
      omit_external_ip: false
      daos_install_type: server
      image_family: daos-server-hpc-rocky-8

- group: slurm
  modules:
  - id: slurm-image
    source: github.com/SchedMD/slurm-gcp//packer?ref=5.7.4&depth=1
    kind: packer
    settings:
      source_image_project_id: cloud-hpc-image-public
      source_image_family: hpc-rocky-linux-8
      use_iap: true
      subnetwork: $(network0.subnetwork_self_link)
      image_family_alt: new-image-family
      slurm_version: 22.05.9
```

### Submission Checklist

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cloud HPC Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

**Comment by tpdownes** [comment]
@mr0re1 please take a careful look at the changes for invoking `packer` shell commands. I believe I am taking the correct approach for capturing stdout/stderr. There is future work planned to intercept ^C and pass it to the packer process. Please consider whether we are doing anything to make that difficult.

File: cmd/deploy.go [hunk 1]
```diff
@@ -89,19 +90,23 @@ func runDeployCmd(cmd *cobra.Command, args []string) error {
...
 		}
 
-		var err error
+		var e error
 		switch group.Kind {
 		case config.PackerKind:
 			// Packer groups are enforced to have length 1
-			moduleDir := filepath.Join(groupDir, string(group.Modules[0].ID))
-			err = deployPackerGroup(moduleDir)
+			subPath, err := modulewriter.DeploymentSource(group.Modules[0])
```
Comment (mr0re1) [inline cmd/deploy.go:97]: nit. This line may create backward compatibility issues in the future (old deployment folder, new ghpc with changed `DeploymentSource` logic). I understand that `filepath.Join(groupDir, string(group.Modules[0].ID))` has exactly the same problem. We may want to explore persisting the mapping `module -> dir` somewhere.

File: cmd/deploy.go [hunk 2]
```diff
@@ -89,19 +90,23 @@ func runDeployCmd(cmd *cobra.Command, args []string) error {
...
+			if err != nil {
```
Comment (mr0re1) [inline cmd/deploy.go:98]: nit. `cobra.CheckErr` to save a couple of lines.

File: modules/packer/custom-image/image.pkr.hcl [hunk 3]
```diff
@@ -70,6 +70,8 @@ locals {
     : local.on_host_maintenance_default
   )
 
+  accelerator_type = var.accelerator_type == null ? null : "projects/${var.project_id}/zones/${var.zone}/acceleratorTypes/${var.accelerator_type}"
```
Comment (mr0re1) [inline modules/packer/custom-image/image.pkr.hcl:73]: Is this change related to the "packaging" ? Reply to this comment is enough, no need to update PR.

Comment (tpdownes) [reply modules/packer/custom-image/image.pkr.hcl:73]: No, it's a separate commit I'm adding because it's related to Packer generally.

**Review by mr0re1 — APPROVED** [review]

File: cmd/deploy.go (hunk 1 above)
Comment (tpdownes) [reply cmd/deploy.go:97]: Would your concern go away if we decided to move Packer modules "up" one level in the directory structure? Probably by eliminating the module ID from the path? We are already enforcing Packer groups to be 1-module long anyhow.

File: cmd/deploy.go (hunk 2 above)
Comment (tpdownes) [reply cmd/deploy.go:98]: Done (in several places)

File: cmd/deploy.go (hunk 1 above)
Comment (tpdownes) [reply cmd/deploy.go]: I am going to close this conversation as PR is approved but we can continue it offline.
